sql: panic in randgen.(*testSchemaGenerator).prepareCounts

roachtest.sqlsmith/setup=empty/setting=no-ddl [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8286064?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/8286064?buildTab=artifacts#/sqlsmith/setup=empty/setting=no-ddl) on master @ [73485e56768d61b14a5aa9b3f880613d7820edb9](https://github.com/cockroachdb/cockroach/commits/73485e56768d61b14a5aa9b3f880613d7820edb9):


```
HINT: node likely crashed, check logs in artifacts > logs/1.unredacted
previous sql:
WITH
	with_4462 (col_24642)
		AS (
			SELECT
				(-1979425290):::INT8 AS col_24642
			FROM
				(VALUES (NULL), ('1997-04-23 18:20:00.000117':::TIMESTAMP)) AS tab_14302 (col_24641)
			WHERE
				true
		),
	with_4465 (col_24655)
		AS (
			SELECT
				*
			FROM
				(
					VALUES
						(
							(('[[[false, true], [true], 0.1466320447506599], 0.23521909922387363, 0.9887345319874119, true, 2.2774876111164852]':::JSONB::JSONB - ((WITH with_4463 (col_24643) AS (SELECT * FROM (VALUES (NULL)) AS tab_14303 (col_24643)), with_4464 (col_24644) AS (SELECT * FROM (VALUES ('':::VOID), ('':::VOID), ('':::VOID), ('':::VOID), (NULL), (NULL)) AS tab_14304 (col_24644)) SELECT '[true, null, "foobar"]':::JSONB AS col_24654 FROM (SELECT ARRAY[(2635693631:::OID, '\x7635':::BYTES, '0bf9d60d-07ef-43ec-9377-3c0acae2ff0b':::UUID, (-4033953550845683798):::INT8),(1009310124:::OID, '\x8706adbaf40a1b':::BYTES, NULL, (-1923089087274825367):::INT8)] AS col_24648, tab_14305.col_24645 AS col_24649, tab_14305.col_24646 AS col_24650 FROM (VALUES ('010500004004000000010200004004000000C68A008E8C46FCC1400840C728EFAD4136C7192B5FE7FFC124F871360CA9E84100C027C84E3C63C14A89AA533FCBEFC19CB1BD305870EA41D40AC443C8D3F641780EB059F72D01C2FCA5FED44E09FC41F202D24350920142BC6B55AF6E54024201020000400500000048A035D39BC3FE4164780A8E2C86FDC17C8D0D995F3C02424819BA7E458FF64133A1918EA153F1C1A0C95F036614CFC198D88A5FA51BFC41706297178D1FD2C12870FB76CFC2EF41BE6A8707326EF94180EFAB5CCCC0B04148AEF0615480E24170CB7EC332D7CDC17A44C24471D8F541408F48907DB4A2C1010200004007000000587CCC61E5E6EB41660CE732B9D4E6C1602D19260AB1B34158336ECC0721D241C4D070025BA0D7C180E12D8AB43AB5C1002080B6E8BCC541F046C8E3453FC0C13838AA4B04BBE54130B90E75CB2EC141E819A5F5AC35EA41F0CA26989466FDC13610186FE3BAE1C15A61BD407A19FA41343F878E1F3FF8C1BAC163826790E8C1A40074DB89ED0142C6129F3D81E1F3416FBE08BC1735F5C1889E05F63993E741282949FD5493FAC1010200004005000000A03723450CDEB8C1C89E1CD798A0FBC1D55AB8CD5AA1F0C1087FA4D234E4D24181971782896FF0C1604A55744CBAEB41040ED528A64CEB4111B1F9FC4071F1C100AED0472CC49641C466AD75F5B8E641D08C3DE97C22EF415DD973AF5F2901C20CB157CECECCEAC1B034A24F3291F341B64C3224DBD6F4C1':::GEOMETRY, 3172435834:::OID, ARRAY[(1098196489:::OID, '\x7d6b46235a8d124fd8':::BYTES, '17e3bc78-f527-4151-951e-aecfd0c5d3e4':::UUID, 4086211756350285411:::INT8),(664526753:::OID, '\x9709801b':::BYTES, 'e050779c-4964-4f82-85bb-706203d0236c':::UUID, 1415689466749507389:::INT8)]), (NULL, 970720331:::OID, ARRAY[(780075556:::OID, '\x1912':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 7830148513575787036:::INT8)]), (NULL, 3390974336:::OID, ARRAY[(3092562690:::OID, '\x1286918ce9792a':::BYTES, NULL, 2064365240179100470:::INT8),(3743701800:::OID, '\x8056192b':::BYTES, '83136ca5-30a0-417d-8e5d-c4b8cf598ec7':::UUID, (-7782534729592890973):::INT8),(4129157252:::OID, '\x398a53b5cffea7':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, (-1657844121818419516):::INT8),(2632165099:::OID, '\xbb12e0641d8b29':::BYTES, '23e27a13-795c-4adc-b173-3d597457b315':::UUID, 2241357000520817282:::INT8)])) AS tab_14305 (col_24645, col_24646, col_24647)) AS tab_14306 (col_24651, col_24652, col_24653) WHERE NULL LIMIT 1:::INT8)::JSONB->>NULL::STRING)::STRING::STRING)::JSONB::JSONB#>CASE WHEN true THEN ARRAY[e'u\x1eUfCW':::STRING,'"':::STRING] ELSE ARRAY[]:::STRING[] END::STRING[])::JSONB
						),
						(
							'{"b": "VcVGwm0", "baz": {"a": [null], "foo": 0.5794506640659327}, "c": [false, 0.22097803144695344, false], "foobar": 2.015430738792313}':::JSONB
						),
						('{"test": "json"}':::JSONB),
						(
							crdb_internal.pb_to_json('':::STRING::STRING, ('\xeebd4fc1ae2fb800':::BYTES::BYTES || NULL::BYTES)::BYTES::BYTES)::JSONB
						),
						(
							crdb_internal.generate_test_objects(e'\U00002603':::STRING::STRING, ARRAY[]:::INT8[]::INT8[])::JSONB
						)
				)
					AS tab_14307 (col_24655)
		),
	with_4466 (col_24656)
		AS (
			SELECT
				*
			FROM
				(
					VALUES
						('21:59:28.794277':::TIME),
						(('62 years 95 days 06:13:21.337458':::INTERVAL::INTERVAL + NULL::TIME)::TIME)
				)
					AS tab_14308 (col_24656)
		)
SELECT
	ARRAY['X':::STRING] AS col_24657;ping node 1: dial tcp 34.23.192.247:26257: connect: connection refused
HINT: node likely crashed, check logs in artifacts > logs/1.unredacted
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=empty/setting=no-ddl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-23367

panic on node 1:
```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 3413 [running]:
panic({0x50b67a0, 0xc00635e150})
	GOROOT/src/runtime/panic.go:987 +0x3ba fp=0xc0072876a8 sp=0xc0072875e8 pc=0x49dd5a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc000cd5000, {0x6b340c0, 0xc001aae580}, {0x52c91e0?, 0xc006d1fb48?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1080 +0x1b4 fp=0xc007287738 sp=0xc0072876a8 pc=0x31f72d4
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x6b340c0, 0xc001aae580}, {0x1ce1a250ce?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:832 +0x4a fp=0xc007287778 sp=0xc007287738 pc=0x31f52ca
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func3()
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:833 +0x32 fp=0xc0072877a0 sp=0xc007287778 pc=0x31f54d2
runtime.deferCallSave(0xc007287870, 0xc00728ddf0?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc0072877b0 sp=0xc0072877a0 pc=0x49d948
runtime.runOpenDeferFrame(0xc00275e910?, 0xc0028414f0)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc0072877f8 sp=0xc0072877b0 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc0072878b8 sp=0xc0072877f8 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc007287a58 sp=0xc0072878b8 pc=0x1bb9707
runtime.deferCallSave(0xc007287b28, 0xc00728b2e8?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc007287a68 sp=0xc007287a58 pc=0x49d948
runtime.runOpenDeferFrame(0xc0028400f0?, 0xc00275e910)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc007287ab0 sp=0xc007287a68 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc007287b70 sp=0xc007287ab0 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc007287d10 sp=0xc007287b70 pc=0x1bb9707
runtime.deferCallSave(0xc007287de0, 0xc00728b218?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc007287d20 sp=0xc007287d10 pc=0x49d948
runtime.runOpenDeferFrame(0xc00275e910?, 0xc0028400f0)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc007287d68 sp=0xc007287d20 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc007287e28 sp=0xc007287d68 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc007287fc8 sp=0xc007287e28 pc=0x1bb9707
runtime.deferCallSave(0xc007288098, 0xc00728b160?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc007287fd8 sp=0xc007287fc8 pc=0x49d948
runtime.runOpenDeferFrame(0xc0028414f0?, 0xc00275e910)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc007288020 sp=0xc007287fd8 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc0072880e0 sp=0xc007288020 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc007288280 sp=0xc0072880e0 pc=0x1bb9707
runtime.deferCallSave(0xc007288350, 0xc0072896d8?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc007288290 sp=0xc007288280 pc=0x49d948
runtime.runOpenDeferFrame(0xc0028400f0?, 0xc00275e910)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc0072882d8 sp=0xc007288290 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc007288398 sp=0xc0072882d8 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x407 fp=0xc007288538 sp=0xc007288398 pc=0x1bb9707
runtime.deferCallSave(0xc007288608, 0xc007289608?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc007288548 sp=0xc007288538 pc=0x49d948
runtime.runOpenDeferFrame(0xc00275e910?, 0xc0028400f0)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc007288590 sp=0xc007288548 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc007288650 sp=0xc007288590 pc=0x49dbb2
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).Generate.func1.1()
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:151 +0x78 fp=0xc007288678 sp=0xc007288650 pc=0x300e338
runtime.deferCallSave(0xc007288748, 0xc007288908?)
	GOROOT/src/runtime/panic.go:796 +0x88 fp=0xc007288688 sp=0xc007288678 pc=0x49d948
runtime.runOpenDeferFrame(0x8?, 0xc00275e910)
	GOROOT/src/runtime/panic.go:769 +0x1a5 fp=0xc0072886d0 sp=0xc007288688 pc=0x49d765
panic({0x52c91e0, 0xc006d1fb48})
	GOROOT/src/runtime/panic.go:884 +0x212 fp=0xc007288790 sp=0xc0072886d0 pc=0x49dbb2
runtime.goPanicIndex(0x0, 0x0)
	GOROOT/src/runtime/panic.go:113 +0x7f fp=0xc0072887d0 sp=0xc007288790 pc=0x49bc1f
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).prepareCounts(0xc010773268?, 0xc007288908?)
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:371 +0x3b4 fp=0xc007288888 sp=0xc0072887d0 pc=0x300e714
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).Generate.func1(0xc0073ddc00, 0x1, {0x6b34168, 0xc0074d9710})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:162 +0xc6 fp=0xc0072889f0 sp=0xc007288888 pc=0x300dca6
github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen.(*testSchemaGenerator).Generate(0xc0073ddc00, {0x6b34168, 0xc0074d9710})
	github.com/cockroachdb/cockroach/pkg/sql/catalog/randgen/randgen.go:212 +0x133 fp=0xc007288a40 sp=0xc0072889f0 pc=0x300da13
github.com/cockroachdb/cockroach/pkg/sql.(*planner).GenerateTestObjects(0xc0074b0000, {0x6b34168, 0xc0074d9710}, {0xc009f1d980, 0x1e})
	github.com/cockroachdb/cockroach/pkg/sql/generate_objects.go:66 +0x5ec fp=0xc007288e20 sp=0xc007288a40 pc=0x330c28c
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.glob..func291({0x6b34168, 0xc0074d9710}, 0xc00697f100, {0xc00b62d680?, 0xc65440?, 0xc007289140?})
	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins.go:6383 +0x90 fp=0xc007288e58 sp=0xc007288e20 pc=0x1c65890
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.(*evaluator).EvalFuncExpr(0xc00697f100, {0x6b34168, 0xc0074d9710}, 0xc00224f000)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:479 +0x118 fp=0xc0072890f0 sp=0xc007288e58 pc=0x18b5698
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FuncExpr).Eval(0xc0074d9230?, {0x6b34168?, 0xc0074d9710?}, {0x6bc59f0?, 0xc00697f100?})
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/sem/tree/eval_expr_generated.go:276 +0x37 fp=0xc007289120 sp=0xc0072890f0 pc=0x13cb717
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.Expr(...)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:26
github.com/cockroachdb/cockroach/pkg/sql.(*valuesNode).startExec(0xc006aaf3b0, {{0x6b34168, 0xc0074d9710}, 0xc00697f100, 0xc0074b0000})
	github.com/cockroachdb/cockroach/pkg/sql/values.go:80 +0x342 fp=0xc007289290 sp=0xc007289120 pc=0x344afa2
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2({0xc007289380?, 0x473284?}, {0x6b37070, 0xc006aaf3b0})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:523 +0x143 fp=0xc007289340 sp=0xc007289290 pc=0x3383703
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112 +0x3e fp=0xc007289378 sp=0xc007289340 pc=0x346431e
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc007289490, {0x6b37070?, 0xc006aaf3b0?}, {0x55c6822?, 0x6?})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297 +0x14c3 fp=0xc007289408 sp=0xc007289378 pc=0x3464223
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc007289490, {0x6b37070, 0xc006aaf3b0})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79 +0xf7 fp=0xc007289458 sp=0xc007289408 pc=0x3462c57
github.com/cockroachdb/cockroach/pkg/sql.walkPlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
github.com/cockroachdb/cockroach/pkg/sql.startExec({{0x6b34168?, 0xc0074d9710?}, 0xc00697f100?, 0xc0074b0000?}, {0x6b37070, 0xc006aaf3b0})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:526 +0x119 fp=0xc0072894d8 sp=0xc007289458 pc=0x3383559
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start(0xc00679bb00, {0x6b34168, 0xc0074d96e0})
	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:173 +0xbd fp=0xc007289518 sp=0xc0072894d8 pc=0x3386cdd
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Init(0xc005f23800, {0x6b34168, 0xc0074d96b0})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:187 +0xea fp=0xc007289550 sp=0xc007289518 pc=0x26d098a
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*panicInjector).Init(0xc0074d9530, {0x6b34168?, 0xc0074d96b0?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/panic_injector.go:62 +0xb7 fp=0xc0072895c8 sp=0xc007289550 pc=0x2f5ee77
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:260 +0x34 fp=0xc0072895f0 sp=0xc0072895c8 pc=0x26d4d94
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc006708960?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc007289630 sp=0xc0072895f0 pc=0x1bb92a2
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start(0xc006708960, {0x6b34168?, 0xc0074d95f0?})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:259 +0x7a fp=0xc007289698 sp=0xc007289630 pc=0x26d4cda
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:122 +0x34 fp=0xc0072896c0 sp=0xc007289698 pc=0x2f5dcd4
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc006be4140?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc007289700 sp=0xc0072896c0 pc=0x1bb92a2
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start(0xc006be4140, {0x6b340c0?, 0xc007412400?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:121 +0x70 fp=0xc007289758 sp=0xc007289700 pc=0x2f5dc50
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc006be4140, {0x6b340c0?, 0xc007412400?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:732 +0x3a fp=0xc007289798 sp=0xc007289758 pc=0x1ee20da
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc00a64d200, {0x6b340c0, 0xc007412400})
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:483 +0x254 fp=0xc007289858 sp=0xc007289798 pc=0x2412754
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc00697ea00?, {0x6b340c0?, 0xc007412400?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:296 +0x205 fp=0xc0072898f8 sp=0xc007289858 pc=0x2f63965
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc0022e03c0, {0x6b34168, 0xc0074d9110}, 0xc009e0fcb0, 0xc003e87ef0, 0xc00224f180, 0xc005b07180, 0xc00697ea00, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:823 +0x95a fp=0xc00728a048 sp=0xc0072898f8 pc=0x32c461a
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc0074d9110?, {0x6b34168, 0xc0074d9110}, 0x6b37070?, 0xc009e0fcb0, 0x0?, {{0x6b37070?, 0xc006aaf3b0?}, 0x0?}, 0xc005b07180, ...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1759 +0x1fb fp=0xc00728a100 sp=0xc00728a048 pc=0x32c8adb
github.com/cockroachdb/cockroach/pkg/sql.runPlanInsidePlan({0x6b34168, 0xc0074d9110}, {{0x6b34168, 0xc0074d9110}, 0xc000cd58b0, 0xc000cd5540}, 0xc007258a80, {0x6b35048?, 0xc001e95000?})
	github.com/cockroachdb/cockroach/pkg/sql/apply_join.go:334 +0x85c fp=0xc00728a8b8 sp=0xc00728a100 pc=0x31c493c
github.com/cockroachdb/cockroach/pkg/sql.(*planner).EvalRoutineExpr.func2(0xc006aaf040, 0xc00728a9d8, {0x6b34168, 0xc0074d8ff0}, 0xc000cd58b0?, {0xc001e94fe0, 0x2, 0x2}, 0xc001e95000, 0xc003e87ef0, ...)
	github.com/cockroachdb/cockroach/pkg/sql/routine.go:99 +0x276 fp=0xc00728a978 sp=0xc00728a8b8 pc=0x33c30f6
github.com/cockroachdb/cockroach/pkg/sql.(*planner).EvalRoutineExpr(0xc000cd5540, {0x6b34168?, 0xc0074d8ff0}, 0xc006aaf040, {0xc001e94fe0?, 0x2, 0x2})
	github.com/cockroachdb/cockroach/pkg/sql/routine.go:105 +0x41c fp=0xc00728aae0 sp=0xc00728a978 pc=0x33c293c
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.(*evaluator).EvalRoutineExpr(0xc00697dc00?, {0x6b34168?, 0xc0074d8ff0?}, 0xc006aaf040?)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:635 +0xaa fp=0xc00728ab50 sp=0xc00728aae0 pc=0x18b66aa
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*RoutineExpr).Eval(0xc00572dad0?, {0x6b34168?, 0xc0074d8ff0?}, {0x6bc59f0?, 0xc00697dc00?})
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/sem/tree/eval_expr_generated.go:346 +0x37 fp=0xc00728ab80 sp=0xc00728ab50 pc=0x13cbe17
github.com/cockroachdb/cockroach/pkg/sql/sem/eval.Expr(...)
	github.com/cockroachdb/cockroach/pkg/sql/sem/eval/expr.go:26
github.com/cockroachdb/cockroach/pkg/sql.(*valuesNode).startExec(0xc006aaf090, {{0x6b34168, 0xc0074d8ff0}, 0xc00697dc00, 0xc000cd5540})
	github.com/cockroachdb/cockroach/pkg/sql/values.go:80 +0x342 fp=0xc00728acf0 sp=0xc00728ab80 pc=0x344afa2
github.com/cockroachdb/cockroach/pkg/sql.startExec.func2({0xc00728ade0?, 0x473284?}, {0x6b37070, 0xc006aaf090})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:523 +0x143 fp=0xc00728ada0 sp=0xc00728acf0 pc=0x3383703
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1()
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:112 +0x3e fp=0xc00728add8 sp=0xc00728ada0 pc=0x346431e
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc00728afd0, {0x6b37070?, 0xc006aaf090?}, {0x55c6822?, 0x6?})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:297 +0x14c3 fp=0xc00728ae68 sp=0xc00728add8 pc=0x3464223
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc00728afd0, {0x6b37070, 0xc006aaf090})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79 +0xf7 fp=0xc00728aeb8 sp=0xc00728ae68 pc=0x3462c57
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal(0xc00728afd0, {0x6b357b8?, 0xc00224ec00?}, {0x55c2aa8?, 0x6?})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:289 +0x107c fp=0xc00728af48 sp=0xc00728aeb8 pc=0x3463ddc
github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit(0xc00728afd0, {0x6b357b8, 0xc00224ec00})
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:79 +0xf7 fp=0xc00728af98 sp=0xc00728af48 pc=0x3462c57
github.com/cockroachdb/cockroach/pkg/sql.walkPlan(...)
	github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
github.com/cockroachdb/cockroach/pkg/sql.startExec({{0x6b34168?, 0xc0074d8ff0?}, 0xc00697dc00?, 0xc000cd5540?}, {0x6b357b8, 0xc00224ec00})
	github.com/cockroachdb/cockroach/pkg/sql/plan.go:526 +0x119 fp=0xc00728b018 sp=0xc00728af98 pc=0x3383559
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start(0xc00679a900, {0x6b34168, 0xc0074d8fc0})
	github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:173 +0xbd fp=0xc00728b058 sp=0xc00728b018 pc=0x3386cdd
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Init(0xc00345ac00, {0x6b34168, 0xc0074d8f90})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:187 +0xea fp=0xc00728b090 sp=0xc00728b058 pc=0x26d098a
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*panicInjector).Init(0xc0074d8de0, {0x6b34168?, 0xc0074d8f90?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/panic_injector.go:62 +0xb7 fp=0xc00728b108 sp=0xc00728b090 pc=0x2f5ee77
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).init(0x946d06?)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:92 +0x2c fp=0xc00728b130 sp=0xc00728b108 pc=0x2f61ccc
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).init-fm()
	<autogenerated>:1 +0x26 fp=0xc00728b148 sp=0xc00728b130 pc=0x2f6d0c6
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc007412240?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc00728b188 sp=0xc00728b148 pc=0x1bb92a2
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Init(0xc00739c1e0, {0x6b34168?, 0xc0074d8f90?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:101 +0x7f fp=0xc00728b1d8 sp=0xc00728b188 pc=0x2f61d7f
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:260 +0x34 fp=0xc00728b200 sp=0xc00728b1d8 pc=0x26d4d94
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc006708780?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc00728b240 sp=0xc00728b200 pc=0x1bb92a2
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Start(0xc006708780, {0x6b34168?, 0xc0074d8f60?})
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:259 +0x7a fp=0xc00728b2a8 sp=0xc00728b240 pc=0x26d4cda
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:122 +0x34 fp=0xc00728b2d0 sp=0xc00728b2a8 pc=0x2f5dcd4
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc00368ef00?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc00728b310 sp=0xc00728b2d0 pc=0x1bb92a2
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Start(0xc00368ef00, {0x6b340c0?, 0xc007412200?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:121 +0x70 fp=0xc00728b368 sp=0xc00728b310 pc=0x2f5dc50
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).Run(0xc00368ef00, {0x6b340c0?, 0xc007412200?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/processorsbase.go:732 +0x3a fp=0xc00728b3a8 sp=0xc00728b368 pc=0x1ee20da
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Run(0xc00a64ce00, {0x6b340c0, 0xc007412200})
	github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:483 +0x254 fp=0xc00728b468 sp=0xc00728b3a8 pc=0x2412754
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc00697d500?, {0x6b340c0?, 0xc007412200?})
	github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:296 +0x205 fp=0xc00728b508 sp=0xc00728b468 pc=0x2f63965
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc0022e03c0, {0x6b34168, 0xc0074d8ab0}, 0xc009e0f9e0, 0xc003e87ef0, 0xc00224ee00, 0xc001708000, 0xc00697d500, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:823 +0x95a fp=0xc00728bc58 sp=0xc00728b508 pc=0x32c461a
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).planAndRunSubquery(0xc0022e03c0, {0x6b34168, 0xc0074d8ab0}, 0x0, {{0x6af5cc0, 0xc006aaeb90}, 0x3, 0x1, 0x0, {{0x6b357b8, ...}, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1635 +0x7c8 fp=0xc00728c098 sp=0xc00728bc58 pc=0x32c7c08
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunSubqueries(0x0?, {0x6b34168, 0xc0074d8ab0}, 0x0?, 0xc00728c3b8, {0xc006aaf180?, 0x1, 0x1}, 0x0?, 0xc00728c2a8, ...)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1543 +0x18c fp=0xc00728c1d8 sp=0xc00728c098 pc=0x32c738c
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc000cd5000?, {0x6b34168, 0xc0074d8ab0}, 0xc00697d500?, 0xc000cd5540?, 0xc000cd5540, 0xc001a40a80, 0xc003e87ef0?)
	github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1500 +0x192 fp=0xc00728c2f0 sp=0xc00728c1d8 pc=0x32c6e72
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc000cd5000, {0x6b34168, 0xc0074d8ab0}, 0xc000cd5540, 0xc0074d8ab0?, {0x7f7b22aa2098?, 0xc000ee0f20}, 0x0?, 0xc004d09a18)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1632 +0x4a8 fp=0xc00728c3f0 sp=0xc00728c2f0 pc=0x3211fe8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc000cd5000, {0x6b34168, 0xc007d8b1a0}, 0xc000cd5540, {0x7f7b22aa2098, 0xc000ee0f20})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1253 +0xc13 fp=0xc00728c820 sp=0xc00728c3f0 pc=0x320f453
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc000cd5000, {0x6b34168?, 0xc007d8b080?}, {{0x6b5d8a0, 0xc006aaecd0}, {0x0, 0x0, 0x0}, {0xc007f9a600, 0x100f}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:738 +0x2654 fp=0xc00728d278 sp=0xc00728c820 pc=0x320a934
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x6b34168?, 0xc007d8b080?})
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:130 +0xbc fp=0xc00728d328 sp=0xc00728d278 pc=0x3207d1c
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0x6b34168?, {0x6b34168?, 0xc007d8b080?}, {0x6b5d8a0?, 0xc006aaecd0?}, 0x0?, 0x9a58bc0?)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:2522 +0x330 fp=0xc00728d478 sp=0xc00728d328 pc=0x3218b70
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc000cd5000, {0x6b34168, 0xc007d8b080}, {{0x6b5d8a0, 0xc006aaecd0}, {0x0, 0x0, 0x0}, {0xc007f9a600, 0x100f}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:129 +0x4fa fp=0xc00728d608 sp=0xc00728d478 pc=0x320771a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0xc00728dac0, 0xc000cd5000, 0xc00728d7b8, 0xc00728d870, 0xc00728d8b0, 0xc00728d890, 0xc00728d880, 0xc00728d8a0)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1959 +0x30f fp=0xc00728d6d0 sp=0xc00728d608 pc=0x31fc1ef
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc000cd5000)
	github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1964 +0x1127 fp=0xc00728dcd0 sp=0xc00728d6d0 pc=0x31fab07
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:640 +0x22e
```
Looks related to your recent change, @knz

--- a/pkg/sql/catalog/randgen/randgen.go
+++ b/pkg/sql/catalog/randgen/randgen.go
@@ -314,7 +314,7 @@ func (g *testSchemaGenerator) prepareCounts(
 	const reasonableMaxCount = 10000000
 	for _, sz := range counts {
 		// We cap the individual count to a maximum number for two purposes.
-		// One is that abnormally large values are likely indicative of
+		// One is that abnormally large values are likely indicative
 		// of a user mistake; in which case we want an informative error message
 		// instead of something related to memory usage.
 		// The second is that we want to avoid an overflow in the computation
@@ -328,8 +328,8 @@ func (g *testSchemaGenerator) prepareCounts(
 	g.gencfg.createDatabases = false
 	g.gencfg.createSchemas = false
 	g.gencfg.useGeneratedPublicSchema = false
-	if len(counts) > 3 {
-		panic(genError{errors.WithHint(pgerror.Newf(pgcode.Syntax, "too many counts"),
+	if len(counts) < 1 || len(counts) > 3 {
+		panic(genError{errors.WithHint(pgerror.Newf(pgcode.Syntax, "invalid count"),
 			"Must specify between 1 and 3 counts.")})
 	}
 	numNewDatabases := 0

--- a/pkg/sql/logictest/testdata/logic_test/gen_test_objects
+++ b/pkg/sql/logictest/testdata/logic_test/gen_test_objects
@@ -568,6 +568,14 @@ SELECT crdb_internal.generate_test_objects('{"names":"a.b.c","counts":[0,0,10000
 query error too many objects generated
 SELECT crdb_internal.generate_test_objects('{"names":"a.b.c","counts":[10000000,10000000,10000000]}'::jsonb)
 
+subtest invalid_count
+
+query error invalid count
+SELECT crdb_internal.generate_test_objects('foo', ARRAY[]::INT8[])
+
+query error invalid count
+SELECT crdb_internal.generate_test_objects('foo', ARRAY[1, 2, 3, 4, 5]::INT8[])
+
 subtest temp_schema
 
 # Force create the temp schema.

